paddle/trainer/ThreadParameterUpdater.cpp

```diff
@@ -141,7 +141,7 @@ void SgdThreadUpdater::traverse(GetTraverseCallback getTraverseCallback) {
   } else if (hasCpuPara) {
     getGlobalSyncThreadPool()->exec(cpuTraverse);
   } else if (hasGpuPara) {
-    cpuTraverse(0, 0);
+    getGlobalSyncThreadPool()->exec(gpuTraverse);
```

Comment:
should be just "gpuTraverse(0,0)", otherwise line131-133 might be executed multiple times.